### DISCUSSION TITLE
The Scope of Primary Key

### DISCUSSION THREAD

**Discussion by W2Q3Q1** [milvus-io/milvus#18202] [Q&A and General discussion]
What is the scope of the primary key? Will their be some data with the same primary key in different partitions in the same collection? Especially when the primary keys are generated automatically?

**Comment by yhmo** [comment]
The data type of the primary key is int64, the range is [-9223372036854775808 ~ 9223372036854775807]
Milvus doesn't check duplicate primary keys, which means you can insert the same primary keys in the same collection.
If you want to ensure each primary key is unique, milvus can help to generate unique primary keys, just define a primary key field like this:
`field = FieldSchema(name=field_name, dtype=DataType.INT64, is_primary=True, auto_id=True)`

**Comment by W2Q3Q1** [comment]
Thanks! But my question is: If auto_id is True, will there be dumplicated ids in different partitions of the same collection? Will there be same id in different collections?



---Original---
From: ***@***.***&gt;
Date: Mon, Jul 11, 2022 10:52 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [milvus-io/milvus] The Scope of Primary Key (Discussion #18202)




 
The data type of the primary key is int64, the range is [-9223372036854775808 ~ 9223372036854775807]
 Milvus doesn't check duplicate primary keys, which means you can insert the same primary keys in the same collection.
 If you want to ensure each primary key is unique, milvus can help to generate unique primary keys, just define a primary key field like this:
 field = FieldSchema(name=field_name, dtype=DataType.INT64, is_primary=True, auto_id=True)

**Reply by yhmo** [reply]
> Thanks! But my question is: If auto_id is True, will there be dumplicated ids in different partitions of the same collection? Will there be same id in different collections?
> […](#)
> ---Original--- From: ***@***.***&gt; Date: Mon, Jul 11, 2022 10:52 AM To: ***@***.***&gt;; Cc: ***@***.******@***.***&gt;; Subject: Re: [milvus-io/milvus] The Scope of Primary Key (Discussion #18202) The data type of the primary key is int64, the range is [-9223372036854775808 ~ 9223372036854775807] Milvus doesn't check duplicate primary keys, which means you can insert the same primary keys in the same collection. If you want to ensure each primary key is unique, milvus can help to generate unique primary keys, just define a primary key field like this: field = FieldSchema(name=field_name, dtype=DataType.INT64, is_primary=True, auto_id=True) — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you authored the thread.Message ID: ***@***.***&gt;

If auto_id is True, milvus can ensure no duplicated ids in a collection, and no duplicated ids in different partitions of the same collection. No duplicated ids in different collections, too.

**Comment by W2Q3Q1** [comment]
Thanks a lot!



---Original---
From: ***@***.***&gt;
Date: Mon, Jul 11, 2022 11:31 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [milvus-io/milvus] The Scope of Primary Key (Discussion #18202)




  
Thanks! But my question is: If auto_id is True, will there be dumplicated ids in different partitions of the same collection? Will there be same id in different collections?
 …
 ---Original--- From: @.&gt; Date: Mon, Jul 11, 2022 10:52 AM To: @.&gt;; Cc: @.@.&gt;; Subject: Re: [milvus-io/milvus] The Scope of Primary Key (Discussion #18202) The data type of the primary key is int64, the range is [-9223372036854775808 ~ 9223372036854775807] Milvus doesn't check duplicate primary keys, which means you can insert the same primary keys in the same collection. If you want to ensure each primary key is unique, milvus can help to generate unique primary keys, just define a primary key field like this: field = FieldSchema(name=field_name, dtype=DataType.INT64, is_primary=True, auto_id=True) — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you authored the thread.Message ID: @.***&gt;
  
If auto_id is True, milvus can ensure no duplicated ids in a collection, and no duplicated ids in different partitions of the same collection. No duplicated ids in different collections, too.
